### ISSUE TITLE
Runtime error with latest build (33b6dcf)

### ISSUE THREAD

**Issue by paulfantom** [google/go-jsonnet#274]
When using latest go-jsonnet downloaded with `go get github.com/google/go-jsonnet/cmd/jsonnet` I am seeing this runtime error:
> INTERNAL ERROR: (CRASH) RUNTIME: Unknown variable: std (we should have caught this statically)

It looks like this was introduced by https://github.com/google/go-jsonnet/pull/263

Full stacktrace: https://travis-ci.org/openshift/cluster-monitoring-operator/jobs/529694792#L666

Strangely in some cases it doesn't produce this error, but then output is different than one created by C++ version of jsonnet binary and different than one created by go-jsonnet before #263.

**Comment by sbarzowski** [comment]
Thanks for reporting, I was able to reproduce with the code from your repository. This change was fairly intrusive so it's very likely it is indeed the culprit. 

> Strangely in some cases it doesn't produce this error, but then output is different than one created by C++ version of jsonnet binary and different than one created by go-jsonnet before #263.

Interesting. Could you expand on what "in some cases" means? How are these cases different from `../../jsonnet -J jsonnet/vendor jsonnet/main.jsonnet`? Is there a way to reproduce this behavior?

**Comment by sbarzowski** [comment]
I was able to reduce the test case to this:
```
{
  local tmp = std.parseHex("ff"), 
  [name]: tmp for name in ["a"]
}
```
So it seems like there is something fishy about the environment used for locals in object comprehensions.

**Comment by paulfantom** [comment]
Those "some cases" are from our other repository, where CI reported additional parts: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_telemeter/169/pull-ci-openshift-telemeter-master-generate/117/build-log.txt

Output is from running `make --always-make && git diff --exit-code` from this repo: https://github.com/openshift/telemeter. It was running with golang 1.10.

**Comment by sbarzowski** [comment]
I've just sent a PR with what I think is a fix (#275). Could you check that it actually works for you? 

(You can get the version from the PR with  `git fetch origin pull/275/head:fix && git checkout fix && go install ./cmd/jsonnet`)

**Comment by sbarzowski** [comment]
Fixed in #275
